File: Data Structures & Algorithms/climbing-stairs/submission-5.cs
public class Solution {
    public int ClimbStairs(int n) {    

        if(n <= 3) 
            return n;
        var n2 = n / 2; // 3 
        var n3 = n2 * (n - n2); // 3 * (6-3) = 6

        return ClimbStairs(n - 1) + ClimbStairs(n - 2);
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        for n in nums: 
            if n not in d:
                d[n] = 0
            d[n] += 1
            
        top_k = list(sorted(d, key=lambda x: d[x], reverse=True))[0:k]
        return top_k
